### DISCUSSION TITLE
How to map custom select results to data class?

### DISCUSSION THREAD

**Discussion by MattSolvesGames** [simolus3/drift#2976] [Q&A] [answered]
I have the following data class:

```dart
@JsonSerializable()
class CardData extends BaseModel {
  final int id;
  final String name;

  CardData(this.id, this.name);
}
```

I have defined a table as such:

```dart
@UseRowClass(CardData)
class Card extends Table {
  IntColumn get id => integer()();
  TextColumn get name => text()();

  @override
  Set<Column> get primaryKey => {id};
}
```

I am building a custom select statement based off user filters. I would like to write a method that returns a list of `CardData` objects. I am under the impression that because I used `@UseRowClass(CardData)` it should be possible to get data from the database as that type but I can't seem to find out how to :

```dart
// buildQuery = "SELECT * FROM card WHERE name = 'test'"

await dbClient.customSelect(buildQuery(filters), readsFrom: {dbClient.card})...  // This gives me a `Selectable<QueryRow>` but how do I map it from here?
```

**Comment by simolus3** [comment, marked as answer]
Drift offers no builtin conversion mechanisms for custom queries (unless they're written in a [drift file](https://drift.simonbinder.eu/docs/sql-api/drift_files/)), but that's something else. If your query is selecting all columns from the table, you can use the `map` method on the generated table:

```dart
dbClient.customSelect(buildQuery(filters), readsFrom: {dbClient.card}).map((row) => dbClient.card.map(row.data))
```
